File: app/src/main/java/com/piyushsatija/pollutionmonitor/SplashActivity.java
package com.piyushsatija.pollutionmonitor;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.os.Handler;
import android.widget.TextView;

public class SplashActivity extends AppCompatActivity {
    private Handler handler = new Handler();

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_splash);
        TextView versionTextView = findViewById(R.id.splash_version_text);
        versionTextView.setText(String.format("v%s", BuildConfig.VERSION_NAME));

        handler.postDelayed(() -> {
            startActivity(new Intent(SplashActivity.this, MainActivity.class));
            finish();
        }, 2000);
    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
        //Remove all the callbacks otherwise navigation will execute even after activity is killed or closed.
        handler.removeCallbacksAndMessages(null);
    }
}

File: app/src/main/java/com/piyushsatija/pollutionmonitor/view/SplashActivity.kt
package com.piyushsatija.pollutionmonitor.view

import android.content.Intent
import android.os.Bundle
import android.os.Handler
import android.view.animation.AccelerateDecelerateInterpolator
import androidx.appcompat.app.AppCompatActivity
import com.piyushsatija.pollutionmonitor.R
import kotlinx.android.synthetic.main.activity_splash.*

class SplashActivity : AppCompatActivity() {
    private val handler = Handler()
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setTheme(R.style.AppTheme_Light)
        setContentView(R.layout.activity_splash)
        splashCloudLeft.animate().setInterpolator(AccelerateDecelerateInterpolator()).translationX(0f).duration = 1500
        splashCloudRight.animate().setInterpolator(AccelerateDecelerateInterpolator()).translationX(0f).duration = 1500
        handler.postDelayed({
            startActivity(Intent(this@SplashActivity, MainActivity::class.java))
            finish()
        }, 2000)
    }

    override fun onDestroy() {
        super.onDestroy()
        //Remove all the callbacks otherwise navigation will execute even after activity is killed or closed.
        handler.removeCallbacksAndMessages(null)
    }
}